### ISSUE TITLE
Allow configuration of STAR parameter --limitGenomeGenerateRAM

### ISSUE THREAD

**Issue by smoe** [BIMSBbioinfo/pigx_rnaseq#83]
It took a while until I noticed that this paramter is nothing that pigx-rnaseq sets but it is a default set by STAR. Somehow, though, the indexing of the mouse genome 38 does not work with the default 31*10^9 but an error message suggests to go for at least 33524373088. I admit not to have any exact idea about what you should do about it. You could obviously try to understand the error message in the logfile and adjust the setting
```
$ less output/logs/star_index.log
Feb 21 20:00:59 ..... started STAR run
Feb 21 20:00:59 ... starting to generate Genome files

EXITING because of FATAL PARAMETER ERROR: limitGenomeGenerateRAM=31000000000is too small for your genome
SOLUTION: please specify --limitGenomeGenerateRAM not less than 33524373088 and make that much RAM available

Feb 21 20:02:00 ...... FATAL ERROR, exiting
```
Maybe to prepare an entry in your rules for that would help? Or you transform the memory information accordingly?

**Comment by borauyar** [comment]
@smoe you can configure tool specific arguments in the settings yaml file. 

tools:
  star_index:
    args: "--limitGenomeGenerateRAM 33524373088"

This should fix the memory issue.

**Comment by smoe** [comment]
Thank you tons for this. Yes, this would have done it. Funnily enough, I did not run into this with mouse genome 39 and was only curious to learn about the differences that the new genome has on the mapping etc - and then this.

To explain that PR a bit - I had first grepped in the pigx-rnaseq source tree and found the parameter in the tests only, then in STAR (has a new version out since a few hours, btw) where I found it as the default. I then added it to the invocation of STAR in the pigx-rnaseq sources. This PR is darn stupid in a way, but it gives the grepping user of yours a target. And yes, I had read through your tutorial. :o)
